src/base/static/client/mapseed-api-client.js

```diff
@@ -58,115 +58,347 @@ const deletePlaceTag = async url => {
-      const includePrivate = hasAdminAbilities(collectionId);
-      // if we are fetching a dataset id to be used in the dashboard,
-      // and the user has access to that dataset:
-      collection.fetchAllPages({
-        remove: false,
-        // Check for a valid location type before adding it to the collection
-        validate: true,
-        data: includePrivate
-          ? {
-              ...placeParams,
-              include_private_places: true,
-              include_private_fields: true,
-            }
-          : placeParams,
-        // get the dataset slug and id from the array of map layers
-        attributesToAdd: {
-          datasetSlug: layer.slug,
-          datasetId: layer.id,
-        },
-        attribute: "properties",
-        xhrFields: { withCredentials: includePrivate },
+  try {
+    const placePagePromises = [];
+    placeParams = includePrivate
+      ? {
+          ...placeParams,
+          include_private_places: true,
+          include_private_fields: true,
+        }
+      : placeParams;
 
-        // Only do this for the first page...
-        pageSuccess: _.once(function(collection, data) {
-          pageSize = data.features.length;
-          totalPages = totalPages += Math.ceil(data.metadata.length / pageSize);
+    const firstPagePromise = fetch(`${url}${buildQuerystring(placeParams)}`, {
+      credentials: "include",
+    })
+      .then(status)
+      .then(json);
 
-          if (data.metadata.next) {
-            $progressContainer.show();
-          }
-        }),
+    placePagePromises.push(firstPagePromise);
 
-        // Do this for every page...
-        pageComplete: function() {
-          var percent;
+    await firstPagePromise.then(data => {
+      // Fetch additional pages of data, if they exist.
+      if (data.metadata.next) {
+        const pageSize = data.features.length;
+        const totalPages =
+          pageSize > 0 ? Math.ceil(data.metadata.length / pageSize) : 0;
 
-          pagesComplete++;
-          percent = (pagesComplete / totalPages) * 100;
-          $currentProgress.width(percent + "%");
+        for (let i = 2; i <= totalPages; i++) {
+          placeParams = { ...placeParams, page: i };
+          placePagePromises.push(
+            fetch(`${url}${buildQuerystring(placeParams)}`, {
+              credentials: "include",
+            })
+              .then(status)
+              .then(json),
+          );
+        }
+      }
+    });
 
-          if (pagesComplete === totalPages) {
-            _.delay(function() {
-              $progressContainer.hide();
-            }, 2000);
-          }
-        },
+    return placePagePromises.map(placePagePromise =>
+      // Convert from GeoJSON to a simple object of Place data.
+      placePagePromise.then(featureCollection =>
+        fromGeoJSONFeatureCollection({
+          featureCollection,
+          datasetSlug,
+          clientSlug,
+        }),
+      ),
```

Comment:
This does look simpler, though wouldn't this remove the ability to load pages of data onto the map as they arrive from the server? I think this is a good usability goal in general to strive for.

Unless the API client also has access to the Redux store, I think it has to return page promises, and the caller has to deal with them as they come, right?

Reply:
>This does look simpler, though wouldn't this remove the ability to load pages of data onto the map as they arrive from the server?

Yes, that's true, except that we aren't loading pages as they arrive. I think this used to be the case, but we removed it. In AppView, you can see the caller is concatenating all of the place pages *then* adding them to the redux store all at once. It's an interesting use case for giving the API client access to the Redux store though :thinking:  But for now, I think it's YAGNI to add this complexity.

Reply:
Ah, that's true. Well, final argument for keeping it: I'm planning to bring back paginated loading in https://github.com/mapseed/platform/pull/1173 (both paginated loading into the duck and as a result paginated loading of places onto the map).

I suppose I could remove the paginated complexity here and add it in #1173 though.

Reply:
That sounds reasonable - feel free to bring it back then :)

So we may want to consider having `getPlaces` return an array of promises, each of which resolves to a places page - instead of a promise that resolves to an array of promises. Up to you - just make sure to clean this up with async/await.

Reply:
Yeah, the Promise of Promises is convoluted-- I'll refactor that.